roles/metrics-export.lua

```diff
@@ -322,15 +322,67 @@ local function routes_equal(old, new)
     return true
 end
 
+local function stop_server(name)
+    local server = (http_servers or {})[name]
+
+    if server ~= nil then
+        if server.httpd_name ~= nil then
+            for path, _ in pairs(server.routes) do
+                server.httpd:delete(path)
+            end
+        else
+            if server.httpd.is_run == true then
+                server.httpd:stop()
+            end
+        end
+
+        http_servers[name] = nil
+    end
+end
+
 local function apply_http(conf)
-    local enabled = {}
+    -- Stop old servers.
+    for name in pairs(http_servers or {}) do
+        local server_found = false
+
+        for _, node in pairs(conf) do
+            if #(node.endpoints or {}) > 0 then
+                local key
+
+                if node.server ~= nil then
+                    key = 'httpd_' .. node.server
+                elseif node.listen ~= nil then
+                    local host, port, err = parse_listen(node.listen)
+                    if err ~= nil then
+                        error("failed to parse URI: " .. err, 2)
+                    end
+
+                    key = 'listen_' .. host .. ':' .. tostring(port)
```

Comment:
Shouldn't we work only with `listen` servers, since metrics-export-role don't start named ones itself?